### PR TITLE
Add test for CDC join scenario

### PR DESCRIPTION
**Pull request by jbartok**
Intention: come up with a pipeline, which reads two different tables from a database via CDC and keeps an up-to-date map with their merged values, in this case customers and their orders. 

Particular details to pay attention to:
* I've added a version of `Sinks.mapWithEntryProcessor` where you can specify `maxParallelAsyncOps`; I have to set that to 1 for this scenario, to avoid Sink-to-IMap reordering due to multiple async requests
* I've changed multiple sinks to have partitioned input-edges, where a partitioning function is naturally available
* the test code probably runs on a local member only, a remote cluster will not know how to deserialize objects like `Customer`, `Order` and `OrdersToCustomer`; if/when we turn this into a tutorial or cookbook entry we will need to change it to work with simpler, standard objects

Point is to prove that we can achieve this joining functionality correctly.

Checklist
- [x] Tags Set
- [x] Milestone Set

### CODE REVIEW

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                            }
+                            return null;
+                        },
+                        (TriFunction<Sequence, RecordPart, Long, ChangeRecord>) (sequence, recordPart, aLong) -> null);
+    }
+
+    @NotNull
+    private static Map<Integer, OrdersOfCustomer> getIMapContent(JetInstance jet, String name) {
+        return jet.<Integer, OrdersOfCustomer>getMap(name).entrySet().stream()
+                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
```
Comment (frant-hartm) [inline extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:157]:
```suggestion
        return new HashMap<>(jet.getMap(name));
```

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+                        record -> (Integer) record.key().toMap().get("id"),
+                        CustomerEntryProcessor::new
+                ));
+
+        allRecords.filter(r -> r.table().equals("orders"))
+                .apply(this::fixOrdering)
+                .writeTo(Sinks.mapWithEntryProcessor(MAX_CONCURRENT_OPERATIONS, CACHE,
+                        record -> (Integer) record.value().toMap().get("purchaser"),
+                        OrderEntryProcessor::new
+                ));
```
Comment (frant-hartm) [inline extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:79]: Will this work when loading larger amount of data from scratch?

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+            return false;
+        }
+
+        @Override
+        public String toString() {
+            return "source=" + source + ", sequence=" + sequence;
+        }
+    }
+
+    private static class OrdersOfCustomer implements Serializable {
```
Comment (frant-hartm) [inline extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:197]: I hoped to have 

```
class Customer {
  Set<Order> orders;
}
```

is there anything blocking us from having this? The `OrdersOfCustomer` is kind of equivalent, but not as nice.

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java (hunk 1 above)
Comment (jbartok) [reply extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:157]: Ok.

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java (hunk 3 above)
Comment (jbartok) [reply extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:197]: We can change it. The thing is though, that: 1) neither the current nor the suggested solution will actually work in a real cluster, due to IMDG not knowing how to deserialize these classes 2) the purpose of this test is to prove that we have SOME way of doing this join correctly and getting the data into IMDG. So... beautifying it should happen probably some place else.

File: extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java (hunk 2 above)
Comment (jbartok) [reply extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java:79]: I don't know.

**Pull request merged**